doc/cprover-manual/goto-harness.md

```diff
@@ -314,87 +314,121 @@ VERIFICATION SUCCESSFUL
 
 ### The memory snapshot harness
 
-The `function-call` harness is used in situations in which we want the analysed
-function to work in arbitrary environment. If we want to analyse a function
-starting from a _real_ program state, we can call the `memory-snapshot` harness
-instead.
-
-Furthermore, the program state of interest may be taken at a particular location
-within a function. In that case we do not want the harness to instrument the
-whole function but rather to allow starting the execution from a specific
-initial location (specified via `--initial-location func[:<n>]`). Note that the
-initial location does not have to be the first instruction of a function: we can
-also specify the _location number_ `n` to set the initial location inside our
-function. The _location numbers_ do not have to coincide with the lines of the
-program code. To find the _location number_ run CBMC with
-`--show-goto-functions`. Most commonly, the _location number_ is the instruction
-of the break-point used to extract the program state for the memory snapshot.
+The `function-call` harness is used in situations in which we want to analyse a
+function in an arbitrary environment. If we want to analyse a function starting
+from a _real_ program state, we can use the `memory-snapshot` harness instead.
+
+The snapshot of the program state of interest may be taken at a particular
+program location within a function (using the `memory-analyzer` tool). In that
+case we want to generate a harness that behaves as if execution starts at a
+particular program location. The initial program location can be specified via
+the options `--initial-goto-location <function>[:<location-number>]` or
+`--initial-source-location <file>:<line-number>`.
 
 Say we want to check the assertion in the following code:
 
 ```C
 // main.c
 #include <assert.h>
+#include <stdlib.h>
```

Comment:
Have you actually run this code to check whether it works?

Reply:
Yes